.circleci/config.yml

```diff
@@ -2,11 +2,10 @@ version: 2.1
 
 references:
   npm_cache_key: &npm_cache_key
-                   v1-dependency-npm1-{{ checksum "package-lock.json" }}
+                   v1-dependency-npm1-{{ checksum "package.json" }}-{{ checksum "package-lock.json" }}
   npm_cache_restore_key: &npm_cache_restore_key
     keys:
       - *npm_cache_key
-      - v1-dependency-npm1
```

Comment:
I guess, we only need `npm_cache_key`, isn't it?

Reply:
The way the CircleCi cache works is it tries to do a string match in the order listed.
The first one is meant to be an exact match and presumably would avoid npm having to download anything. All the packages are in the cache.
The next entry in the list of keys is meant for an inexact match. npm will have to download stuff, but hopefully not every single package. 
With the ci flag, the second entrywill take a bit of thought. Perhaps it is all or nothing with the ci flag?

Reply:
Thanks for the explanation!
I think it would be better if we restrict the cache version to be sure that we use correct packages versions. Moreover, we don't change packages versions very often and `npm ci` works fast, so we can install deps if we didn't find a cache.